src/utils/detectHooks.js

```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+export default function detectHooks() {
+  return React.useContext && React.useRef && React.useEffect && React.useState;
```

Comment:
@ljharb Is checking one of them sufficient?

Reply:
probably so, but checking all of them also can't hurt.